frontend/src/stories/TextInput.stories.tsx

```diff
@@ -0,0 +1,24 @@
+import { Story, Meta } from "@storybook/react";
+
+import { TextInput } from "../app/commonComponents/TextInput";
+
+export default {
+  title: "Components/Text input",
+  component: TextInput,
+  argTypes: {},
+  args: {
+    name: "text-input",
+    label: "Text input",
+    errorMessage: "Helpful error message",
+  },
+} as Meta;
+
+const Template: Story = (args) => <TextInput {...args} />;
```

Comment:
The typescript compiler is yelling because `TextInput` expects at least a name, label and onChange prop but we have given it no assurances that `args` will contain those props. The following change is a reusable way for future stories as well to tell typescript that that `args` will have the same type signature as the component.

```suggestion
type Props = React.ComponentProps<typeof TextInput>;

const Template: Story<Props> = (args) => <TextInput {...args} />;
```

Reply:
Thanks! For some reason, the `WithError` story didn't work after this, so I removed it (we can still toggle the state in the default story)